local-tests/tests/testSignTransactionWithSolanaEncryptedKey.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  Connection,
+  Keypair,
+  LAMPORTS_PER_SOL,
+  PublicKey,
+  SystemProgram,
+  Transaction,
+  clusterApiUrl,
+} from '@solana/web3.js';
+import {
+  LitAccessControlConditionResource,
+  LitActionResource,
+  createSiweMessageWithRecaps,
+  generateAuthSig,
+} from '@lit-protocol/auth-helpers';
+
+/**
+ * Test Commands:
+ * ✅ NETWORK=cayenne yarn test:local --filter=testSignTransactionWithSolanaEncryptedKey
+ * ✅ NETWORK=manzano yarn test:local --filter=testSignTransactionWithSolanaEncryptedKey
+ * ❔ NETWORK=localchain yarn test:local --filter=testSignTransactionWithSolanaEncryptedKey
+ */
+export const testSignTransactionWithSolanaEncryptedKey = async (
+  devEnv: TinnyEnvironment
+) => {
+  const alice = await devEnv.createRandomPerson();
+  const solanaKeypair = Keypair.generate();
+  const decryptionAccessControlConditions: AccessControlConditions = [
+    {
+      contractAddress: '',
+      standardContractType: '',
+      chain: 'ethereum',
+      method: '',
+      parameters: [':userAddress', 'latest'],
+      returnValueTest: {
+        comparator: '=',
+        value: await alice.wallet.getAddress(),
+      },
+    },
+  ];
+
+  const { ciphertext, dataToEncryptHash } = await encryptString(
+    {
+      accessControlConditions: decryptionAccessControlConditions,
+      dataToEncrypt: Buffer.from(solanaKeypair.secretKey).toString('hex'),
+    },
+    devEnv.litNodeClient as unknown as ILitNodeClient
+  );
+
+  const sessionSigs = await devEnv.litNodeClient.getSessionSigs({
+    chain: 'ethereum',
+    resourceAbilityRequests: [
+      {
+        resource: new LitActionResource('*'),
+        ability: LitAbility.LitActionExecution,
+      },
+      {
+        resource: new LitAccessControlConditionResource('*'),
+        ability: LitAbility.AccessControlConditionDecryption,
+      },
+    ],
+    authNeededCallback: async ({ uri, resourceAbilityRequests }) => {
+      const toSign = await createSiweMessageWithRecaps({
+        uri: uri!,
+        expiration: new Date(Date.now() + 1000 * 60 * 10).toISOString(),
+        resources: resourceAbilityRequests!,
+        walletAddress: await alice.wallet.getAddress(),
+        nonce: await devEnv.litNodeClient!.getLatestBlockhash(),
+        litNodeClient: devEnv.litNodeClient,
+      });
+      return await generateAuthSig({
+        signer: alice.wallet,
+        toSign,
+      });
+    },
+  });
+
+  const solanaTransaction = new Transaction();
+  solanaTransaction.add(
+    SystemProgram.transfer({
+      fromPubkey: solanaKeypair.publicKey,
+      toPubkey: new PublicKey(solanaKeypair.publicKey),
+      lamports: LAMPORTS_PER_SOL / 100, // Transfer 0.01 SOL
+    })
+  );
+  solanaTransaction.feePayer = solanaKeypair.publicKey;
+
+  const solanaConnection = new Connection(clusterApiUrl('devnet'), 'confirmed');
+  const { blockhash } = await solanaConnection.getLatestBlockhash();
+  solanaTransaction.recentBlockhash = blockhash;
```

Comment:
Is it mandatory? What if we don't provide the blockhash?

Reply:
Yes it's mandatory, the blockhash is used to ensure the transaction is processed within a specific timeframe and helps prevent replay attacks. Without it the tx is invalid